### DISCUSSION TITLE
Optional Schema Definition in Eloquent Models

### DISCUSSION THREAD

**Discussion by bikify** [laravel/framework#48425] [Ideas]
Ok

**Comment by ghost** [comment]
Put this in your model's construct

`$this->getConnection()->setDatabaseName('custom_name');`

or
`$this->getConnection()->setDatabaseName($this::DB_SCHEMA);`

or
`$this->getConnection()->setDatabaseName(static::DB_SCHEMA);`

and define the DB_SCHEMA constant in your model.

UPDATE
now I saw the previous text.... My answer is not for that :))
